src/OrchardCore.Modules/OrchardCore.Forms/Views/ContentCard-Button.Inline.cshtml

```diff
@@ -0,0 +1,19 @@
+@using OrchardCore.ContentManagement.Metadata.Settings;
+@inject OrchardCore.ContentManagement.Metadata.IContentDefinitionManager ContentDefinitionManager
+
+@{
+    var contentItem = (IContent)Model.ContentItem;
+    var contentTypeDisplayText = ContentDefinitionManager.GetTypeDefinition((string)Model.ContentItem.ContentType).DisplayName;
+    var contentItemDisplayText = contentItem.ContentItem.DisplayText;
+
+    var btnText = ((ContentItem)Model.ContentItem).As<OrchardCore.Forms.Models.ButtonPart>().Text;
```

Comment:
I think somehow this `ContentCard` needs to become driver based / or produced by a driver.
There are too many overrides / special alternates here, no one will understand how it works or how to extend it.

Reply:
I guess keep ContentCard simple  with blank shape that provides only Header, Content and Footer.

- If you want some custom shape to be injected in toolbar call `cardShape.Footer.Add(yourtoolbarshape)`
- If you want shape to be displayed in body of Card then call `cardShape.Content.Add(yourbodyshape)`
- If you want shape to be displayed in Header of Card then call `cardShape.Header.Add(yourheadershape)`
- If you don't want to render Card body then set Content to null `cardShape.Content = null;`

Reply:
Sorry it's taken me so long to get back to this @ns8482e 

I've been working through a couple of other prs, and needed to have a chat with Seb, to see which design works better.

Here's the two prs
- Bad design https://github.com/OrchardCMS/OrchardCore/pull/6249
- Good design https://github.com/OrchardCMS/OrchardCore/pull/6551

In the first I used many shape providers to affect the layout, and items in the zones.

In the refactor I have moved all of the shape providers to using an `IDisplayManager`.
This meant the drivers, where able to provide the correct views, for the correct zones, with none of the issues that you are seeing with shape providers here.

So my suggestion is to consider creating a model for your `ContentCardModel` and driving the shapes you want into it by calling an `IDisplayManager<ContentCardModel>`, in the BagPart and FlowPart edit views.

What do you think?

(it might require a core project of some sort, because the widgets uses the `ContentCard` as well I think?)

This is quite a big pr, and hard to review, because it has so many things in it, so it might be easier, if we were able to do the content card work on one pr, and improvements to the Forms on another

Reply:
@deanmarcussen I agree with splitting on multile PR's - I was working on `forms` but then fall into domino's effect of changes, I'll work as you suggested.

Reply:
Great thanks @ns8482e I fall into the same domino's all the time too :)

Reply:
> So my suggestion is to consider creating a model for your `ContentCardModel` and driving the shapes you want into it by calling an `IDisplayManager<ContentCardModel>`, in the BagPart and FlowPart edit views.

@deanmarcussen I am confused here, How could I leverage `IDisplayManager` when Editor shape is created via `BuildEditor` without changing the build editor function as ContentCard is Parent shape of Shape created by BuildEditor.

Reply:
@ns8482e I will try and push something experimental to another branch tomorrow so you can see what I am suggesting

Reply:
@ns8482e I pushed a concept of how I would see it working with an `IDisplayManager` to this branch for you to have a look at and see what you think https://github.com/OrchardCMS/OrchardCore/tree/deanmarcussen/content-card-refactor 

I have made it work for the `FlowPart`, but only for existing flow widgets, not adding items (it would, but it's just an idea).

There is much I had to guess about the `ContentCard` so have left comments on bits I am confused on.

See if it makes sense? Skype me or get me on gitter if you want to chat about it

Reply:
@deanmarcussen I'll have look. 

But here with `ContentCard `- My goal was not to introduce any placement specific for `ContentCard` instead have  an event (  `OnDisplay` )  that allow module developer to change UI  for `ContentCard` - by default it uses same placement  as Editor shapes e.g. `Footer` of editor is displayed in Footer of  `ContentCard` and Editor shape is the `Content` of `ContentCard` . 

Keeping it simple such that `Content Type` can be edited with or without `Flow` and use same placement. I doubt having it strongly typed - may be it less usable in `liquid` too